cpp/src/fil/treelite_import.cu

```diff
@@ -349,69 +356,53 @@ int tree2fil(std::vector<fil_node_t>& nodes,
              std::size_t* leaf_counter,
              cat_sets_owner* cat_sets)
 {
-  typedef std::pair<int, int> pair_t;
-  std::stack<pair_t> stack;
   // needed if the node is sparse, to place within memory for the FIL tree
-  int sparse_index = root + 1;
-  stack.push(pair_t(tree_root(tree), 0));
-  while (!stack.empty()) {
-    const pair_t& top = stack.top();
-    int node_id       = top.first;
-    int cur           = top.second;
-    stack.pop();
-
-    while (!tree.IsLeaf(node_id)) {
+  int sparse_index = 1;
+  walk_tree<int>(
+    tree,
+    [&](int node_id, int fil_node_id) {
       // reserve space for child nodes
       // left is the offset of the left child node relative to the tree root
       // in the array of all nodes of the FIL sparse forest
-      int left = node_traits<fil_node_t>::IS_DENSE ? 2 * cur + 1 : sparse_index - root;
+      int left = node_traits<fil_node_t>::IS_DENSE ? 2 * fil_node_id + 1 : sparse_index;
```

Comment:
sparse_index should just be relative to tree, not forest. renames `cur` to `fil_node_id` because otherwise I kept forgetting what it was.

Reply:
This is fine.